### DISCUSSION TITLE
Ask For Help:How to define the specified URL

### DISCUSSION THREAD

**Discussion by mayjack0312** [openstyles/stylus#1526] [Q&A] [answered]
# Problem description

I want to add a style to a specific address
```https://www.baidu.com/```
But at the same time, I want to block
```https://www.baidu.com/s?wd=*```
How can I shield addresses with the same prefix and different endings?
I modified it for half a day, but it can only be applied at the same time.

### CSS Code

I have tried:

```@-moz-document regexp("https?://(www|images|scholar).baidu.(com|([a-z]{2}))(.[a-z]{2})?."), regexp("https?://(www|images|scholar).baidu.(com|([a-z]{2}))(.[a-z]{2})?/((\\?|[a-z]|webhp|gfe_rd|auth|gws_rd|#q|imghp|#gfe_rd|#safe|#pws|#tbs|#gws|#tbm).*)")```

### System Information

- OS: Windows10 22H2 19045.2364
- Browser: Chrome 108.0.5359.99
- Stylus Version: 1.5.28

**Comment by tophf** [comment, marked as answer]
Use `?!` (the negative lookahead): `regexp("https://www\\.baidu\\.com/(?!s\\?wd=).*")`

**Reply by mayjack0312** [reply]
> Use `?!` (the negative lookahead): `regexp("https://www\\.baidu\\.com/(?!s\\?wd=).*")`

 Thank you. I'll try it
